hypertrace-trace-enricher/src/main/java/org/hypertrace/traceenricher/trace/enricher/TraceEnricher.java

```diff
@@ -1,26 +1,96 @@
 package org.hypertrace.traceenricher.trace.enricher;
 
+import static org.hypertrace.traceenricher.trace.enricher.StructuredTraceEnricherConstants.ENRICHER_CONFIGS_KEY;
+import static org.hypertrace.traceenricher.trace.enricher.StructuredTraceEnricherConstants.ENRICHER_CONFIG_TEMPLATE;
+import static org.hypertrace.traceenricher.trace.enricher.StructuredTraceEnricherConstants.ENRICHER_NAMES_CONFIG_KEY;
+import static org.hypertrace.traceenricher.trace.enricher.StructuredTraceEnricherConstants.INPUT_TOPIC_CONFIG_KEY;
+import static org.hypertrace.traceenricher.trace.enricher.StructuredTraceEnricherConstants.JOB_CONFIG;
+import static org.hypertrace.traceenricher.trace.enricher.StructuredTraceEnricherConstants.KAFKA_STREAMS_CONFIG_KEY;
+import static org.hypertrace.traceenricher.trace.enricher.StructuredTraceEnricherConstants.OUTPUT_TOPIC_CONFIG_KEY;
+
 import com.typesafe.config.Config;
-import org.hypertrace.core.serviceframework.background.PlatformBackgroundJob;
-import org.hypertrace.core.serviceframework.background.PlatformBackgroundService;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Produced;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.kafkastreams.framework.KafkaStreamsApp;
 import org.hypertrace.core.serviceframework.config.ConfigClient;
+import org.hypertrace.core.serviceframework.config.ConfigUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class TraceEnricher extends PlatformBackgroundService {
-  private static Logger LOGGER = LoggerFactory.getLogger(TraceEnricher.class);
+public class TraceEnricher extends KafkaStreamsApp {
+
+  private static final Logger logger = LoggerFactory.getLogger(TraceEnricher.class);
 
   public TraceEnricher(ConfigClient configClient) {
     super(configClient);
   }
 
   @Override
-  protected PlatformBackgroundJob createBackgroundJob(Config config) {
-    return new StructuredTraceEnrichmentJob(config);
+  public StreamsBuilder buildTopology(Map<String, Object> properties, StreamsBuilder streamsBuilder,
+      Map<String, KStream<?, ?>> inputStreams) {
+    String inputTopic = getAppConfig().getString(INPUT_TOPIC_CONFIG_KEY);
```

Comment:
What's the use of passing `properties` as argument if everything can be fetched via `getAppConfig()`?

Reply:
For now, we are fetching job config from properties (HOCON structure), this will be useful in hypertrace-ingester. After a first cut of e2e, we will see if we can simplify this one too.